app/services/middleware_topology_service.rb

```diff
@@ -26,39 +26,36 @@ def build_topology
     topology[:kinds] = build_kinds
   end
 
-  def build_entity_data(entity, kind)
-    status = entity_status(entity, kind)
-    id = case kind
-         when 'MiddlewareDeployment', 'MiddlewareServer' then entity.nativeid
-         when 'Hawkular' then entity.id.to_s
-         else entity.ems_ref
-         end
-
-    {:id => id, :name => entity.name, :status => status, :kind => kind, :miq_id => entity.id}
+  def entity_display_type(entity)
+    if entity.kind_of?(ManageIQ::Providers::MiddlewareManager)
+      entity.type.split('::')[2]
```

Comment:
@abonas Aaah, that's it, thanks... well, I would hope for a more descriptive solution, I see `split('::')[2]` is already used in a bunch of places but it's neither clean nor descriptive. Maybe define something like `.provider_name` on the classes so we can rewrite to `entity.class.provider_name`? Anyone?